trillian/ctfe/serialize.go

```diff
@@ -15,24 +15,54 @@
 package ctfe
 
 import (
+	"bytes"
 	"crypto"
 	"crypto/rand"
 	"crypto/sha256"
 	"fmt"
+	"sync"
 
 	"github.com/google/certificate-transparency-go/tls"
 
 	ct "github.com/google/certificate-transparency-go"
 )
 
-// signV1TreeHead signs a tree head for CT. The input STH should have been built from a
-// backend response and already checked for validity.
-func (li *logInfo) signV1TreeHead(signer crypto.Signer, sth *ct.SignedTreeHead) error {
+// SignatureCache stores the last generated signature for a given bytes input.
```

Comment:
This now reads as though the number of entries is going to grow over time if the input changes. It still only holds one entry? Might be worth adding that to the doc.